pkg/compactdictionary/compactdictionary.go

```diff
@@ -41,34 +40,27 @@ func CompactDictionary(mem memory.Allocator, arr *array.Dictionary) (*array.Dict
 	newLen := 0
 	keepValues := make([]int, arr.Dictionary().Len())
 
-	var indices interface{}
-
-	switch arr.Indices().(type) {
+	switch indices := arr.Indices().(type) {
 	case *array.Int32:
-		indices = arr.Indices().(*array.Int32)
-
-		for i := 0; i < indices.(*array.Int32).Len(); i++ {
+		for i := 0; i < arr.Len(); i++ {
 			if arr.IsValid(i) {
-				if keepValues[indices.(*array.Int32).Value(i)] == 0 {
+				if keepValues[indices.Value(i)] == 0 {
 					// keep track of how many values we need to keep to reserve the space upfront
 					newLen++
 				}
-				keepValues[indices.(*array.Int32).Value(i)]++
```

Comment:
this is very important and is missing in both switch cases

Reply:
I have no idea why but the linter/precommit keeps removing this line